src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,224 @@
+using System;
+using System.Text;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+using Windows.Storage;
+using System.Collections.Generic;
+
+namespace Windows.Storage
+{
+	public partial class FileIO
+	{
+
+		private static async Task AppendWriteTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding, bool append)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			Encoding encodingForWriter = Encoding.UTF8;
+			switch (encoding)
+			{
+				case Streams.UnicodeEncoding.Utf8:
+					encodingForWriter = Encoding.UTF8;
+					break;
+				case Streams.UnicodeEncoding.Utf16LE:
+					encodingForWriter = Encoding.Unicode;
+					break;
+				case Streams.UnicodeEncoding.Utf16BE:
+					encodingForWriter = Encoding.BigEndianUnicode;
+					break;
+			}
+
+			Stream fileStream = null;
+			try
+			{
+				fileStream = await file.OpenStreamForWriteAsync();

```

Comment:
Why did you implement the using pattern manually ?

This should work better:
```suggestion
				using(var fileStream = await file.OpenStreamForWriteAsync())

```

Reply:
Because Codacy shows error when `using` is inside `using`.
https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4827590

Reply:
Please, choose either
a) `using` inside `using`, or
b) `using` inside `try/finally`,
and if you choose (a), please block this codacy rule.